modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -226,6 +227,18 @@
   protected ExecutionMediator executionMediator;
   protected CompletableComponentExecutor componentExecutor;
   protected ReturnDelegate returnDelegate;
+  /**
+   * TODO: MULE-18483 When a policy is applied to an operation that has defined a target, it's necessary to wait until the policy
+   * finishes to calculate the return value with {@link #returnDelegate}. But in this case, because of in order to execute the
+   * rest of the policy we need to transform the {@link Result} returned by the operation into a {@link CoreEvent}, we use
+   * {@link #valueReturnDelegate} as a helper class to do this transformation. It's used only when there is an operation that
+   * defines a target, and at the same time, there are operation policies applied to it. Finally, when the policy finishes, the
+   * proper {@link #returnDelegate} is executed.
+   * <p>
+   * It's an horrible solution, we only need a piece of code that transforms an {@link Object} into a {@link CoreEvent} and not a
```

Comment:
rephrase, and point to the fact that the piece of code we need is already in the ValueReturnDelegate, we just need to extract it